src/main/java/nl/knmi/geoweb/backend/product/taf/TafValidator.java

```diff
@@ -852,42 +853,32 @@ private static void augmentEndTimes(JsonNode input) throws ParseException {
 	}
 
 	private static void augmentAscendingClouds(JsonNode input) throws ParseException {
-		
-
-		
 		List<JsonNode> forecasts = input.findParents("clouds");
 		for (JsonNode forecast : forecasts) {
 			if (forecast == null || forecast.isNull() || forecast.isMissingNode())
 				continue;
-			ObjectNode editableForecast = (ObjectNode) forecast;
 			int prevHeight = 0;
-			JsonNode node = forecast.findValue("clouds");
+			ArrayNode node = (ArrayNode) forecast.findValue("clouds");
```

Comment:
This cast fails when clouds is a textnode (like "clouds": "NSC" in Taf_invalid.json").
Suggested fix:
```
JsonNode cloudNode=forecast.findValue("clouds");
if (node.getClass()......{..} //Keep this check
ArrayNode node=(ArrayNode) node; //Code only gets here if not a TextNode